src/init.cpp

```diff
@@ -135,7 +157,8 @@ void updateConfigurationValue(const pika::program_options::variables_map& vm, T&
   const std::string dlaf_env_var = "DLAF_" + env_var;
   char* env_var_value = std::getenv(dlaf_env_var.c_str());
   if (env_var_value) {
-    var = parseFromString<T>::call(env_var_value);
+    if (auto parsed_value = parseFromString<T>::call(env_var_value))
+      var = parsed_value.value();
```

Comment:
We can decide to add a warning to the user here in the `else` branch.

Reply:
I would add the warning if the variable exists but an incorrect value is set.

Reply:
Yes, exactly.

The first `if` checks variable existence. If it exists, we try to parse it with `parseFromString<T>` function that returns an `optional`:

- if it contains a value, it is because we were able to parse it (and understand it)
- otherwise (the suggested `else` branch) we warn the user that something went wrong.

Reply:
What I was suggesting is to already add the else branch with the warning.

Reply:
See https://github.com/eth-cscs/DLA-Future/pull/913/commits/f87cc0c38f155346e4a7fd1f43bfbef5d5eb97b0